src/pages/WorkspaceSwitcherPage/index.tsx

```diff
@@ -87,7 +89,16 @@ function WorkspaceSwitcherPage() {
             setActiveWorkspaceID(newPolicyID);
             Navigation.goBack();
             if (newPolicyID !== activeWorkspaceID) {
-                Navigation.navigateWithSwitchPolicyID({policyID: newPolicyID});
+                // On native platforms, we will see a blank screen if we navigate to a new HomeScreen route while navigating back at the same time.
+                // Therefore we delay switching the workspace until after back navigation, using the InteractionManager.
+                const platform = getPlatform();
+                if (platform === CONST.PLATFORM.IOS || platform === CONST.PLATFORM.ANDROID) {
```

Comment:
We are really trying to to use platform specific code like this, could you add this to native.index.tsx file?

Reply:
updated! 👍